Sources/Vehicle/Templates/CarPlayTemplateProvider.swift

```diff
@@ -0,0 +1,10 @@
+import CarPlay
+import Foundation
+
+protocol CarPlayTemplateProvider {
+    var template: CPTemplate { get set }
```

Comment:
might not be a bad idea to make this (if you can) `some CPTemplate` (and `any CarPlayTemplateProvider`) so you can specialize each usage of it, so they don't have to cast internally

Reply:
Good one!